itwin-platform-access/imodels-access-backend/src/BackendIModelsAccess.ts

```diff
@@ -265,15 +304,21 @@ export class BackendIModelsAccess implements BackendHubAccess {
       checkpoint = await this._iModelsClient.checkpoints.getSingle(getSingleCheckpointParams);
     } catch (error) {
       // Means that neither v1 nor v2 checkpoint exists
-      if (isIModelsApiError(error) && error.code === IModelsErrorCode.CheckpointNotFound)
+      if (isIModelsApiError(error) && error.code === IModelsErrorCode.CheckpointNotFound) {
+        if (arg?.allowPreceding)
```

Comment:
Two `if`s are slightly hard to read here, maybe a ternary operator would look better? Same in line 318

Reply:
changed it, and also opened new PR